src/super_gradients/training/pipelines/pipelines.py

```diff
@@ -266,31 +266,83 @@ def __init__(
+
+        :param target_bboxes_format: Optional[str], bounding box format of target_bboxes, one of ['xyxy','xywh',
+        'yxyx' 'cxcywh' 'normalized_xyxy' 'normalized_xywh', 'normalized_yxyx', 'normalized_cxcywh']. Will raise an
+        error if not None and target_bboxes is None.
+
         :return:                Predicted Bboxes.
         """
+        target_bboxes, target_class_ids = self._check_target_args(target_bboxes, target_bboxes_format, target_class_ids)
+
         post_nms_predictions = self.post_prediction_callback(model_output, device=self.device)
+        if target_bboxes is None:
+            target_bboxes = [None for _ in range(len(model_input))]
+            target_class_ids = [None for _ in range(len(model_input))]
 
         predictions = []
-        for prediction, image in zip(post_nms_predictions, model_input):
+        for prediction, image, target_bbox, target_class_id in zip(post_nms_predictions, model_input, target_bboxes, target_class_ids):
             prediction = prediction if prediction is not None else torch.zeros((0, 6), dtype=torch.float32)
+            target_bbox = target_bbox if target_bbox is not None else np.zeros((0, 4))
+            target_class_id = target_class_id if target_class_id is not None else np.zeros((0, 1))
             prediction = prediction.detach().cpu().numpy()
+
             predictions.append(
                 DetectionPrediction(
                     bboxes=prediction[:, :4],
                     confidence=prediction[:, 4],
                     labels=prediction[:, 5],
                     bbox_format="xyxy",
+                    target_bboxes=target_bbox,
+                    target_labels=target_class_id,
+                    target_bbox_format=target_bboxes_format,
                     image_shape=image.shape,
                 )
             )
 
         return predictions
 
+    @staticmethod
+    def _check_target_args(
+        target_bboxes: Optional[Union[np.ndarray, List[np.ndarray]]] = None,
+        target_bboxes_format: Optional[str] = None,
+        target_class_ids: Optional[Union[np.ndarray, List[np.ndarray]]] = None,
+    ):
+        if (
+            (target_bboxes is None and target_bboxes_format is not None)
+            or (target_bboxes is not None and target_bboxes_format is None)
+            or (target_class_ids is None and (target_bboxes is not None or target_bboxes_format is not None))
+            or (target_class_ids is not None and (target_bboxes is None or target_bboxes_format is None))
+        ):
+            raise ValueError("target_bboxes, target_bboxes_format, and target_class_ids should either all be None or all not None.")
+
+        if isinstance(target_bboxes, np.ndarray):
+            target_bboxes = [target_bboxes]
+        if isinstance(target_class_ids, np.ndarray):
+            target_class_ids = [target_class_ids]
+
+        return target_bboxes, target_class_ids
```

Comment:
please check that all list are of same length

Reply:
added in 16bd98c744f224f04057a99af2dfc48ee77acc96